.github/workflows/ci.yml

```diff
@@ -33,15 +36,64 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - run: rustup default ${{ matrix.rust }}
-      # TODO: MOVE CARGO BUILD TEST TO NEW BUILD JOB & CHECK FOR ANY BUILD WARNINGS TO BE RESOLVED
-      - run: cargo build --features alloc ${{ matrix.test-options }} --verbose
       - run: cargo test --features alloc ${{ matrix.test-options }} --verbose
-      # TODO: MOVE CARGO BUILD TEST TO NEW BUILD JOB & CHECK FOR ANY BUILD WARNINGS TO BE RESOLVED
-      - run: cargo build --all-features ${{ matrix.test-options }} --verbose
-        if: (!startsWith(matrix.os, 'windows'))
       - run: cargo test --all-features ${{ matrix.test-options }} --verbose
         if: (!startsWith(matrix.os, 'windows'))
 
+  build:
+    strategy:
+      # XXX TBD ??? ???
+      fail-fast: false
+      matrix:
+        include:
+          - rust: nightly
+          - os: macos-latest
+            rust: nightly
+          - os: windows-latest
+            rust: nightly
+          - rust: nightly-2024-07-01
+          - rust: nightly-2023-07-01
+          - rust: nightly-2022-08-24 # MSRV (nightly)
+    runs-on: ${{ matrix.os || 'ubuntu-latest' }}
+    steps:
+      - uses: actions/checkout@v4
+      - run: rustup default ${{ matrix.rust }}
+      - run: cargo build --features alloc --verbose
+        env:
+          # XXX TODO RESOLVE REMAINING WARNINGS (OR IGNORE ELSEWHERE)
+          RUSTFLAGS: --cfg portable_io_unstable_all --deny warnings --allow unexpected-cfgs
+      - run: cargo build --all-features --verbose
+        if: (!startsWith(matrix.os, 'windows'))
+        env:
+          # XXX TODO RESOLVE REMAINING WARNINGS (OR IGNORE ELSEWHERE)
+          RUSTFLAGS: --cfg portable_io_unstable_all --deny warnings --allow unexpected-cfgs
+
+  clippy:
+    strategy:
+      # XXX TBD ??? ???
+      fail-fast: false
+      matrix:
+        include:
+          - rust: nightly
+          - os: macos-latest
+            rust: nightly
+          # XXX TBD SKIP FOR NOW (AT LEAST):
+          # - os: windows-latest
+          #   rust: nightly
+    runs-on: ${{ matrix.os || 'ubuntu-latest' }}
+    steps:
+      - uses: actions/checkout@v4
+      - run: rustup default ${{ matrix.rust }}
+      - run: rustup component add clippy
+      # XXX TODO REMOVE THESE
+      - run: rustc --version
+      - run: cargo --version
+      - run: cargo clippy --version
+      - run: cargo clippy --all-features
+        env:
+          # XXX TODO RESOLVE REMAINING WARNINGS (OR IGNORE ELSEWHERE)
+          RUSTFLAGS: --cfg portable_io_unstable_all --deny warnings --allow clippy::redundant-guards --allow clippy::needless-lifetimes --allow clippy::mem-replace-with-default --allow clippy::explicit-auto-deref --allow unexpected-cfgs --allow clippy::unnecessary-cast
+
```

Comment:
I am thinking to hold off on adding `clippy` CI job for now. I would like to try updating from a much newer version of the Rust std library code, as discussed in #19, before attempting to add & configure Clippy with whatever may need to be ignored with these updates.

```suggestion
```